packages/angular/src/tracing.ts

```diff
@@ -241,3 +286,24 @@ export function TraceMethodDecorator(): MethodDecorator {
     return descriptor;
   };
 }
+
+function getParamsOfRoute(activatedRouteSnapshot: ActivatedRouteSnapshot): Params {
+  let params = {};
+  const stack: ActivatedRouteSnapshot[] = [activatedRouteSnapshot];
+  while (stack.length > 0) {
+    const route = stack.pop();
+    params = { ...params, ...(route && route.params) };
+    route && stack.push(...route.children);
```

Comment:
Wooo DFS. Nit: I prefer recursion when walking trees like this to better enforce immutability, but the stack is fine.

also let’s please use the explicit `if (route)`